### PR TITLE
#13 first level

### PR DESCRIPTION
**Pull request by zoompower**

### CODE REVIEW

File: Hivemind/Assets/SpatialPartition.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/SpatialPartition.cs]: Put this file in the scripts folder pls

Comment (ErikLouwerse) [reply Hivemind/Assets/SpatialPartition.cs]: Also remove unused usings

File: Hivemind/Assets/SpatialPartition.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    // Start is called before the first frame update
+    void Start()
+    {
+        for (int i = 0; i < width; i++)
+        {
+            for (int j = 0; j < depth; j++)
+            {
+                GameObject go = Instantiate(obj, transform, false);
+                go.transform.position = new Vector3(
+                    (0.0f - transform.localScale.x / 2.0f) + ((transform.localScale.x / width) / 2.0f) + (transform.localScale.x / width) * i,
```
Comment (ErikLouwerse) [inline Hivemind/Assets/SpatialPartition.cs]: Lots of pointless brackets, please run some sort of auto-cleaner to get rid of this and pointless usings at the same time

File: Hivemind/Assets/Scripts/SpatialPartitioning.cs [hunk 3]
```diff
@@ -43,10 +44,21 @@ public List<GameObject> GetEntitiesWithNeigbors()
     {
         List<GameObject> EntitiesWithNeighbors = Entities;
 
+        List<SpatialPartitioning> Test = new List<SpatialPartitioning>();
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/SpatialPartitioning.cs]: No need to initialize a list if you are going to set it to a list, initialization is only needed if you are going to Add etc.

**Comment by MMCQ33** [comment]
Nav Mesh agents are missing when spawned in (old map)(Generated navmesh is wrong)
Ants dont have a mind in the new map because they never arrive at base
also no ingame ui in level 1 therefore the ants dont have any behaviour. (pls fix)
Moving camera also not present in level 1

File: Hivemind/Assets/Scripts/SpatialPartitioning.cs [hunk 4]
```diff
@@ -56,20 +68,24 @@ public List<GameObject> GetEntitiesWithNeigbors()
 
     public List<GameObject> GetEntitiesWithExtraNeighbors(int steps)
     {
-        List<GameObject> EntitiesWithExtraNeighbors = Entities;
+        List<GameObject> EntitiesWithExtraNeighbors = new List<GameObject>();
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/SpatialPartitioning.cs]: See other comment about initialization

File: Hivemind/Assets/Scripts/SpatialPartitioning.cs [hunk 5]
```diff
@@ -56,20 +68,24 @@ public List<GameObject> GetEntitiesWithNeigbors()
...
 
-        for (int i = -steps; i<steps;i++) 
+        for (int i = -steps; i <= steps; i++)
         {
-            for (int j = -steps; j < steps; j++)
+            for (int j = -steps; j <= steps; j++)
             {
-                GameObject NeigGameObject = GameObject.Find("CollisionBox(" + (height + i) + "," + (width + j) + ")");
-                if (NeigGameObject != null)
+                var Testing = transform.parent.Find("CollisionBox(" + (height + i) + "," + (width + j) + ")");
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/SpatialPartitioning.cs]: Recommend interpolated string for things like this for readability
`transform.parent.Find($"CollisionBox({height + i},{width + j})");`

File: Hivemind/Assets/Scenes/Tests/SpatialPartitioningTests.cs [hunk 6]
```diff
@@ -0,0 +1,99 @@
+﻿using System.Collections;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scenes/Tests/SpatialPartitioningTests.cs]: pointless usings cleanup

File: Hivemind/Assets/Scenes/Tests/SpatialPartitioningTests.cs [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+            var GoFrom = Floor.transform.Find("CollisionBox(3,3)");
+            var GoTo = Floor.transform.Find("CollisionBox(3,5)");
+
+            SpatialPartitioning GoFromSelected = GoFrom.GetComponent<SpatialPartitioning>();
+            SpatialPartitioning GoToSelected = GoTo.GetComponent<SpatialPartitioning>();
+
+            GoFromSelected.Entities.Add(new GameObject());
+            GoToSelected.Entities.Add(new GameObject());
+
+            List<GameObject> Entities = new List<GameObject>();
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scenes/Tests/SpatialPartitioningTests.cs]: See comment about initializing lists, same goes for the 2 in the methods below

File: Hivemind/Assets/Scenes/Map.unity [hunk 8]
```diff
@@ -106,15 +106,15 @@ NavMeshSettings:
   m_BuildSettings:
     serializedVersion: 2
     agentTypeID: 0
-    agentRadius: 0.03
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scenes/Map.unity]: Why was this changed? Map no longer works because of this

File: Hivemind/Assets/Scenes/Tests/SpatialPartitioningTests.cs (hunk 6 above)
Comment (ErikLouwerse) [reply Hivemind/Assets/Scenes/Tests/SpatialPartitioningTests.cs]: Also why is this in the Scenes folder? Seems to fit better in the Scripts folder

**Review by ErikLouwerse — CHANGES_REQUESTED** [review]

File: Hivemind/Assets/Scenes/Tests/SpatialPartitioningTests.cs (hunk 6 above)
Comment (zoompower) [reply Hivemind/Assets/Scenes/Tests/SpatialPartitioningTests.cs]: Test map already existed, unity has its Tests there by default

File: Hivemind/Assets/SpatialPartition.cs (hunk 2 above)
Comment (zoompower) [reply Hivemind/Assets/SpatialPartition.cs]: Those brackets are there for readability reasons, I feel like it would be best to leave those in to make it easier to work with. They separate calculations from each other so you know what is calculated "localy"

File: Hivemind/Assets/Scenes/Map.unity (hunk 8 above)
Comment (zoompower) [reply Hivemind/Assets/Scenes/Map.unity]: Good question. I don't remember editing that specifically

Comment (ErikLouwerse) [reply Hivemind/Assets/Scenes/Map.unity]: Please revert this then, if this were merged develop would no longer work

**Review by ErikLouwerse — APPROVED** [review]

**Review by jetenergy — APPROVED** [review]
